includes/admin/class-sensei-onboarding.php

```diff
@@ -34,18 +53,27 @@ class Sensei_Onboarding {
 	 */
 	public $pages;
 
+	/**
+	 * REST API for Onboarding.
+	 *
+	 * @var Sensei_Onboarding_API
+	 */
+	public $api;
+
 	/**
 	 * Sensei_Onboarding constructor.
 	 */
 	public function __construct() {
 
 		$this->page_slug = 'sensei_onboarding';
 		$this->pages     = new Sensei_Onboarding_Pages();
+		$this->api       = new Sensei_Onboarding_API( $this );
```

Comment:
If we move to controllers, probably we should initialize and register them all at a higher level, rather than at the level of the Onboarding functionality.

Reply:
I'd generally prefer to group things that logically belong together, eg. parts of the same feature, rather than by what kind of things they are. This makes the code easier to work with, as these files are usually worked on together, and they do actually reference eachother at a code level. 

In this case it's especially true, as this is not a public REST API that we provide for some part of the application, but a core part of the feature that connects the frontend and backend bits.

Reply:
Hmm. Generally I agree, and it's a common way to set up JS code. Based on my experience, though, this is not as conventional for WP PHP code, and certainly would seem like a big departure from how the PHP codebase is set up in Sensei.

@jom any thoughts on this?

Reply:
In WordPress (and other PHP projects I've worked on) it does seem convention to group the REST API as a central namespace, but I get what you're saying about this being a special endpoint. I think part of it is around versioning, which doesn't matter as much for internal endpoints, and part of it is around the convention of just initializing REST API controllers inside of `rest_api_init` action.